src/hooks/useANJBalanceToUsd.js

```diff
@@ -0,0 +1,101 @@
+import { useEffect, useState } from 'react'
+import {
+  TRADE_EXACT,
+  getMarketDetails,
+  getTokenReserves,
+  getTradeDetails,
+} from '@uniswap/sdk'
+import { useCourtConfig } from '../providers/CourtConfig'
+
+import env from '../environment'
+import { getDaiAddress } from '../utils/known-tokens'
+import { bigNum, formatUnits } from '../lib/math-utils'
+import { addressesEqual, ETH_FAKE_ADDRESS } from '../lib/web3-utils'
+
+const UNISWAP_MARKET_RETRY_EVERY = 1000
+
+async function getANJMarketDetails(tokenAddress, anjTokenAddress) {
+  const [tokenData, anjData] = await Promise.all(
+    [tokenAddress, anjTokenAddress].map(address => {
+      // In the case of ETH, undefined should be passed as the reserves data.
+      if (addressesEqual(address, ETH_FAKE_ADDRESS)) {
+        return undefined
+      }
+
+      const tokenReserves = getTokenReserves(address, env('CHAIN_ID'))
+
+      if (!tokenReserves) {
+        throw new Error('Could not fetch reserves')
+      }
+      return tokenReserves
+    })
+  )
+
+  return getMarketDetails(tokenData, anjData)
+}
+
+async function getANJTradeDetails(tokenAddress, anjTokenAddress, tradeAmount) {
+  const anjMarketDetails = await getANJMarketDetails(
+    tokenAddress,
+    anjTokenAddress
+  )
+
+  return getTradeDetails(TRADE_EXACT.OUTPUT, tradeAmount, anjMarketDetails)
+}
+
+export default function useANJBalanceToUsd(amount) {
```

Comment:
Ah, so there's generally some confusion over how Uniswap and its SDK work in this regard.

This hook is getting the converted **execution** amount if a real trade happened right now with ANJ<>DAI on the given amount (`getTradeDetails()`).

Since we're not interested in doing the actual conversion, we don't care about slippage (or liquidity, or any of the other Uniswap concerns)—we only care about the **spot price**. That can be obtained via just `getMarketDetails()`, and its `.marketRate`. We then multiply the spot price with the ANJ amount given to get the converted value.